src/fiatExchanges/FiatExchangeAmount.tsx

```diff
@@ -153,17 +155,33 @@ function FiatExchangeAmount({ route }: Props) {
       currency,
       flow,
     })
-
-    navigate(Screens.SelectProvider, {
-      flow,
-      selectedCrypto: currency,
-      amount: {
-        crypto: inputCryptoAmount.toNumber(),
-        // Rounding up to avoid decimal errors from providers. Won't be
-        // necessary once we support inputting an amount in both crypto and fiat
-        fiat: Math.round(inputLocalCurrencyAmount.toNumber()),
-      },
-    })
+    const cryptoAmount = inputCryptoAmount.toNumber()
+    // Rounding up to avoid decimal errors from providers. Won't be
+    // necessary once we support inputting an amount in both crypto and fiat
+    const fiatAmount = Math.round(inputLocalCurrencyAmount.toNumber())
```

Comment:
aren't these subject to floating point and/or rounding errors? Seems risky to do this casting

Reply:
Yes, we have a ticket to go through the CICO flow and fix these https://app.zenhub.com/workspaces/acquisition-squad-sprint-board-6010683afabec1001a090887/issues/valora-inc/wallet/2461